### ISSUE TITLE
Is it possible to style elements that are not assigned an html element or class?

### ISSUE THREAD

**Issue by mark8044** [daohoangson/flutter_widget_from_html#556]
I have a strange use case where my code text is as follows:

```
Some text lorem ipsun
<img class="myimage" src="... an image .." />
Some more text
```

This is not formatted with any other div or classes, but I could if needed place a div to surround the entire block (text and image)

What I am trying to do is then place this block into a container.  
- I want to give that container some padding so that the text isn't on the edge of the container.  
- But I want the image to extend all the way to the edge of the container

Ive come up with 2 strategies to try to do this:

### Strategy 1 ###
Give the flutter Container widget a padding
```
return Container(
             padding:EdgeInsets.all(30),
             child: HtmlWidget(text ...)
);
```
The problem here is that its impossible to get the image to the outer edge of the container (as far as I can figure)

### Strategy 2 ###
Give the text an intrinsic margin via `customStylesBuilder`

I change the text to surround it with a div of class `textContainer` as follows:

```
<div class="textContainer">
Some text lorem ipsun
<img class="myimage" src="... an image .." />
Some more text
</div>
```

Then 

```
   return HtmlWidget(text,
      textStyle: TextStyle(
        color:Color.fromRGBO(0, 0, 0, 1),
        fontSize:15
      ),
      factoryBuilder: () => _WidgetFactory(),
      onTapUrl: (url) => _URLhandler(url),
      customStylesBuilder: (element) {

        if (element.classes.contains('textContainer')) {
          return {
            'position' : 'relative',
            'margin-left': '30px',
            'padding-right': '30px',
          };
        }
}
      customWidgetBuilder: (element) {

        if (element.classes.contains('myimage')) {
          return imgWidget(element.text.toString());
        }
}
```

This successfully gives my text spacing, but the image still does not goto the edge because its a child of the `textContainer` div.  

Using negative margins on the `img` does not seem to work

### Strategy 3 ###
I was thinking going back to my original string
```
Some text lorem ipsun
<img class="myimage" src="... an image .." />
Some more text
```

I could actually just apply some styling to the text that's not part of any div (in this case the margin/padding I want), thereby allowing the `img` to stay without any CSS styling and would go flush with my outer container

- I was thinking strategy 3 would be viable if I could give styling to it without being part of a div (thereby avoiding the nesting problem)
- Is there a better strategy to approach this?  
- Is there a way I can pull the image out of the tree in the sense that I don't want it to follow the `textContainer` CSS rules?

**Comment by daohoangson** [comment]
A `display: block` element will be rendered alone so you can force image to be block then apply padding to the text. See this example:

<details>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_widget_from_html/flutter_widget_from_html.dart';

const kHtml = '''
Some text lorem ipsun
<img class="myimage" src="https://via.placeholder.com/2048x1152" />
Some more text
''';

class Issue556 extends StatelessWidget {
  @override
  Widget build(BuildContext context) => Scaffold(
        appBar: AppBar(title: Text('Issue 556')),
        body: SingleChildScrollView(
          child: HtmlWidget(
            kHtml,
            factoryBuilder: () => _WidgetFactory(),
          ),
        ),
      );
}

class _WidgetFactory extends WidgetFactory {
  @override
  void parse(BuildMetadata meta) {
    super.parse(meta);

    if (meta.element.classes.contains('myimage')) {
      meta['display'] = 'block';
    }
  }

  @override
  Widget buildText(BuildMetadata meta, TextStyleHtml tsh, InlineSpan text) {
    var built = super.buildText(meta, tsh, text);

    if (built != null) {
      built = Padding(padding: const EdgeInsets.all(8), child: built);
    }

    return built;
  }
}

```

</details>


<img width="300" alt="Screen Shot 2021-08-07 at 09 49 42" src="https://user-images.githubusercontent.com/239336/128585589-64fe72b2-838b-4b6e-9aa8-0742fa0bd48a.png">

The important bits were:

```dart
if (meta.element.classes.contains('myimage')) {
  meta['display'] = 'block';
}

// and...

built = Padding(padding: const EdgeInsets.all(8), child: built);
```

**Comment by mark8044** [comment]
Thanks!!

Im understanding more and more....

Right now I'm having an interaction with a quote parser for which I made a separate BuildOp.  Now what is happening is that quote block is no longer falling into line with the now padded text, so the quote block falls to touch the edge of my container

So to fix this I simply added the text padding to that as well

```
    if (meta.element.classes.contains('bbcode_quote')) {

      meta.register(BuildOp(
        onWidgets: (meta, widgets) => listOrNull(
          buildColumnPlaceholder(meta, widgets)?.wrapWith(
                (_, child) {

                  debugPrint(meta.parentOps.toString());
                  return _TextPadding(_quoteWidget(child), textPadding);
                }
          ),
        ),
      ));
    }
```

Which did realign my quote block with the text.

However, in the case where the quotes are nested with each other (the other issue I opened yesterday and you gracefully helped me with!), the nested children are each getting a lot of text padding as well.

is it possible to have that text padding only apply on the root element and not the nested elements?
